File: app/src/main/java/com/wesai/kotlin/test/TestJava.java
package com.wesai.kotlin.test;


import android.content.Context;
import android.support.v4.view.NestedScrollingParent;
import android.widget.LinearLayout;


/**
 * Created by long on 2017/11/16.
 */

class TestJava extends LinearLayout implements NestedScrollingParent {

    public TestJava(Context context) {
        super(context);
        t(null);

    }

    public <T> void t(Class<T> mClass) {

    }
}

File: app/src/main/java/com/wesai/kotlin/test/TestJava.kt
package com.wesai.kotlin.test


/**
 * Created by long on 2017/11/16.
 */

internal class TestJava @JvmOverloads constructor(name: String, age: Int, like: String = "") {
    //    IPresenterFactory<V, P> p = null;
    //
    //    public void test() {
    ////        var pFactory:IPresenterFactory<V, P> =BasePresenterFactory.getFactory(javaClass);
    //
    //        p = BasePresenterFactory.getFactory(getClass());
    //    }
    //
    //    public P getP() {
    //        return p.createPresenter();
    //    }


    constructor(name: String) : this("", 10) {}


}
